### ISSUE TITLE
large amount of queries in a short time will cause "Connection acquisition timed out" error

### ISSUE THREAD

**Issue by seoker** [jamesfer/cypher-query-builder#32]
When I tried to execute a bunch of queries in a very short time, a `Neo4jError` will be thrown.

After investigation, the issue is caused by excessive open connection since every `run()` create a new session to execute which would create new socket connection.

Might consider using a session pool, or just make it possible to use a specified session.

**Comment by jamesfer** [comment]
Interesting idea. I did some investigating and it seems that the underlying neo4j driver already uses a connection pool to limit the number of sessions ([here](https://github.com/neo4j/neo4j-javascript-driver/blob/bfa41c164491fd371938a89c0439ab311801a391/src/v1/driver.js#L172)).

Did the change in your fork fix the problem? Or could you show me what you were running to cause the problem so I could investigate further?

**Comment by seoker** [comment]
Always using the same session might not a good idea since it obliterate the ability of parallelism execution so that I close my PR.

I did encountered `Neo4jError` if I execute a bunch of queries in a very short time, e.g. import data from a large CSV file. (well, `LOAD CSV` is not that convenient because I have to put my CSV file in some specific place). Wondering if cypher builder will take care the session just retrieved? Seems like it exceeding some threshold if we just create sessions and never close it.

Haven't extracted from my code yet, but below snippet might able to address this issue (sorry I've not verified yet)
```javascript
for (let i = 0; i < 10000; ++i) {
  db.createNode('test', 'Test', { value: i }).run();
}
```

**Comment by jamesfer** [comment]
Closing this as the issue would probably be solved by #106.
